### PR TITLE
Josh Grillo all tasks done

### PR DESCRIPTION
**Pull request by L-ux**

### CODE REVIEW

**Comment by L-ux** [comment]
Some of the changes in the files don't have previews, but it all works. Any way to sort this?

File: Bezier.cpp [hunk 1]
```diff
@@ -4,9 +4,77 @@
...
+
+	// set percentage to 0, the start of the bezier curve
+	float percent = 0;
+
+	// generate many points for the bezier curve, according to precision
+	for (int i = 0; i < precision+2; i++)
+	{
+		points[i] = FindPosition(percent);
+
+		percent += 1.0 / precision;
```
Comment (edpowley) [inline Bezier.cpp:16]: This could cause floating point errors to accumulate as the loop progresses. Calculating `percent = static_cast<float>(i) / static_cast<float>(precision)` on each iteration of the loop could avoid this.

File: Bezier.cpp [hunk 2]
```diff
@@ -4,9 +4,77 @@
...
+	}
+
+	// now we gonna try making this path THICC
+
+	for (int j = 0; j < precision+1; j++)  // OUTER LINE
+	{
+		int x_dif = points[i+1].x - points[i].x;  
+		int y_dif = points[i+1].y - points[i].y;
```
Comment (edpowley) [inline Bezier.cpp:24]: Looks like you're on the right lines here -- however the fact that you're using `int` for everything means you're prematurely rounding. You should be using `float`.

File: Bezier.cpp [hunk 3]
```diff
@@ -4,9 +4,77 @@
...
+		SDL_RenderDrawLineF(renderer, points[i].x, points[i].y, points[i + 1].x, points[i + 1].y);
+		SDL_RenderDrawLineF(renderer, outerLine[i].x, outerLine[i].y, outerLine[i + 1].x, outerLine[i + 1].y);
+
+	}
+
+}
+
+Vector2 Bezier::GetVectorPosition(float percent)
+{
+	Vector2 vectorPosition = Vector2();
```
Comment (edpowley) [inline Bezier.cpp:54]: Slight inefficiency here -- technically you're creating a new `Vector2` and copying it. You could just write
```
Vector2 vectorPosition;
```
for the same effect.

File: Bezier.cpp [hunk 4]
```diff
@@ -4,9 +4,77 @@
...
+
+	vectorPosition = pow((1 - percent), 3)*p0 + 3 * pow((1 - percent), 2)*percent*p1 + 3 * (1 - percent) * pow(percent, 2) * p2 + pow(percent, 3) * p3;
```
Comment (edpowley) [inline Bezier.cpp:56]: `pow` might be less efficient than multiplication here

File: Bezier.cpp [hunk 5]
```diff
@@ -4,9 +4,77 @@
...
+
+SDL_Point Bezier::ConvertVector2(Vector2 vector)
+{
+	SDL_Point point = SDL_Point();
+
+	point.x = vector.x;
+	point.y = vector.y;
+
+	return point;
+}
```
Comment (edpowley) [inline Bezier.cpp:69]: Might it be better to put this method somewhere other than the `Bezier` class?

**Review by edpowley — COMMENTED** [review]
Comments on Application.cpp:

- `modifier` isn't a very descriptive variable name -- also some description of how it is calculated would be useful
- Note that `Vector2` has a `magnitude` method that you could use e.g. on line 79
- You're using `3.1412` as an approximation for pi -- this is a little off, pi to 4 decimals is 3.1416. However the `M_PI` constant is available, which gives a much more accurate value and should be used for readability

In summary, well done for completing all the tasks in a mostly clear and efficient manner, and well done also for attempting making the track thicker!
